src/MassTransit/Configuration/BatchOptions.cs

```diff
@@ -33,6 +33,11 @@ public BatchOptions()
         /// </summary>
         public TimeSpan TimeLimit { get; set; }
 
+        /// <summary>
+        /// The condition upon which to group messages in different batches
+        /// </summary>
+        public Func<ConsumeContext, object> GroupingExpression { get; set; }
```

Comment:
to be honest, I do not like `object` idea...

Reply:
I didn't see much value in typing the return value of the grouping expression, because it is only used as the key for `CurrentConsumersDictionary`.